articles/mysql-query-performance-optimization-tips/index.md

```diff
@@ -0,0 +1,180 @@
+```
+```sql
+CREATE INDEX tablename_column1name_column2name_idx ON tablename (column1name, column2name);
+```
+
+#### Use Wildcards at the End of a Phrase
+In MySQL, wildcards are used in conjunction with the LIKE operator and NOT LIKE operator. They are used to search for data matching some search criteria. You can learn more about wildcards [here](https://www.guru99.com/wildcards.html).
+
+Wildcards result in the most expansive scan when searching for data, which is very inefficient. Leading wildcards are the most inefficient, especially when combined with ending wild cards. In such a case, MySQL has to search all the records for a match. Thus you should avoid leading wild cards. See the queries bellow, one using a leading wildcard and another one using ending wildcards.
+
+```sql
+SELECT * FROM city WHERE name LIKE '%Al%';
+```
+
+```sql
+SELECT * FROM city WHERE name LIKE 'Al%';
+```
+
+#### Specify Columns in SELECT Function
+SELECT * (select all) is used as a shortcut to query all columns available in a table.  This requires more resources than using a SELECT statement with only the columns you need for that specific query. For example, a customer table with 20 different columns and a hundred thousand entries. If you want to select a city with ID and Name only; use
+
+```sql
+SELECT ID, Name, District FROM city;
+```
+
+instead of
+
+```sql
+SELECT * FROM city;
+```
+
+The second example will take more time to run to completion.
+
+#### Avoid SELECT DISTINCT
+DISTINCT is used to remove duplicate rows with SELECT statements. The DISTINCT command requires more sorting and reading of the database, which requires more processing power. DISTINCT can be replaced with GROUP BY to get the same results. See the two queries below.
+
+```sql
+SELECT col1, col2 FROM table GROUP BY col1, col2;
+```
+
+```sql
+SELECT DISTINCT col1, col2 FROM table;
+```
+
+#### Use LIMIT
+Sometimes we need a specified number of rows from a result set. LIMIT clause is used in the query to specify the number of rows instead of fetching the whole result set. Fetching the entire result set requires more resources compared to fetching a specified number of rows. See the queries below, one without LIMIT, another one with the LIMIT clause.
+
+```sql
+SELECT ID, Name, District FROM city;
+```
+
+```sql
+SELECT ID, Name, District FROM city LIMIT 10;
+```
+
+#### MySQL Query Caching
+MySQL Query Caching provides database caching functionality. The SELECT statement text and the retrieved result are stored in the cache. When you make a similar query to the one already in the cache, MySQL will respond and give a query already in the cache. In this way, fewer resources are used, and your query runs faster.
+
+This works best with a database the more select queries are made. Once the table is updated, the cached query and result become invalid. Thus, caching may not work with an application that updates the table frequently.
```

Comment:
This sentence is cumbersome, consider revising for more readability: "This works best with a database the more select queries are made."